src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrStates = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
```suggestion
  for (const action of actions) {
  const { type, extraData, keysToRemove } = action;
    switch (type) {
    ...
```
Always use desctruction